### PR TITLE
Report kafka config without source

### PR DESCRIPTION
**Pull request by NetanelBollag**
### Description

Current implementation generate the source field in the backend, so the operator shouldn't report it

- [x] This change adds test coverage for new/changed/fixed functionality

### CODE REVIEW

**Comment by orishoshan** [comment]
Just to clarify - the source field that was previously there is an implementation detail left over from an old API, and as the new API does not have it, it was removed here.

File: src/operator/controllers/kafkaserverconfig_controller_test.go [hunk 1]
```diff
@@ -134,6 +135,10 @@ func (s *KafkaServerConfigReconcilerTestSuite) reconcile(namespacedName types.Na
 		res, err = s.reconciler.Reconcile(context.Background(), ctrl.Request{
 			NamespacedName: namespacedName,
 		})
+
+		if err != nil && strings.Contains(err.Error(), "the object has been modified; please apply your changes to the latest version and try again") {
```
Comment (orishoshan) [inline src/operator/controllers/kafkaserverconfig_controller_test.go:139]: use `if k8serrors.IsConflict` instead - see examples elsewhere

**Review by orishoshan — APPROVED** [review]
fix comment then merge

File: src/operator/controllers/kafkaserverconfig_controller_test.go (hunk 1 above)
Comment (NetanelBollag) [reply src/operator/controllers/kafkaserverconfig_controller_test.go]: Did it

**Review by orishoshan — APPROVED** [review]

**Pull request merged**